### PR TITLE
32 bit sequencer hash

### PR DESCRIPTION
**Pull request by no2chem**
This PR is a slight improvement over #829. It uses a string consisting
of the hash code plus the stream ID as a key, we will have to monitor
it carefully to see if this causes additional GC pressure or performance
regression.

### CODE REVIEW

File: infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java [hunk 1]
```diff
@@ -236,7 +236,7 @@ public TokenType txnCanCommit(TxResolutionInfo txInfo, /** Input. */
                 // for each key pair, check for conflict;
                 // if not present, check against the wildcard
                 conflictParamSet.forEach(conflictParam -> {
-                    int conflictKeyHash = getConflictHashCode(entry.getKey(),
+                    String conflictKeyHash = getConflictHashCode(entry.getKey(),
```
Comment (medhavidhawan) [inline infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java]: Looks fine but it is not solving Anny's case. Do you plan to introduce the bigger hash in the next PR.
Also do you need more tests as the hashcode is now a string ?

File: infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java [hunk 2]
```diff
@@ -193,8 +193,8 @@ public SequencerServer(ServerContext serverContext) {
     * @param conflictParam The conflict parameter.
     * @return A conflict hash code.
     */
-    public int getConflictHashCode(UUID streamId, int conflictParam) {
-        return Objects.hash(streamId, conflictParam);
+    public String getConflictHashCode(UUID streamId, int conflictParam) {
+        return streamId.toString() + Integer.toString(conflictParam);
```
Comment (dahliamalkhi) [inline infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java]: I think that by default, toStrong() prints the Java hashcode of the object, so this brings us back to where we started, no? 

if the intention is that this is just one step towards the full solution, I buy it.

Comment (no2chem) [inline infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java]: The hashcode Java uses here doesn't matter what matters is during a hashcode collision equals will be called and the correct conflict param will be used, unlike before. I'll write a test which shows this.

**Comment by corfudb-performance** [comment]
Results automatically generated by _CorfuDB Benchmark Framework_ to assess the performance of this pull request for commit **710d657**.

*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 1 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 5 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 10 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Iterator workload, 1 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Iterator workload, 5 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Iterator workload, 10 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 1 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 5 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 10 threads, Disk mode
 
 An interactive dashboard with Pull Request Performance Metrics for ALL cluster types and numbers of threads in run, is available at:
 [Pull Request #833 Graphs ](https://snapshot.raintank.io/dashboard/snapshot/oVDfTzX0htR1sLX2uJaP3Iz4cf5YVbjC)

File: infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java (hunk 1 above)
Comment (no2chem) [inline infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java]: It does solve the situation described in the unit test added.

**Comment by corfudb-performance** [comment]
Results automatically generated by _CorfuDB Benchmark Framework_ to assess the performance of this pull request for commit **373a742**.

*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 1 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 5 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 10 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Iterator workload, 1 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Iterator workload, 5 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Iterator workload, 10 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 1 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 5 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 10 threads, Disk mode
 
 An interactive dashboard with Pull Request Performance Metrics for ALL cluster types and numbers of threads in run, is available at:
 [Pull Request #833 Graphs ](https://snapshot.raintank.io/dashboard/snapshot/A62xRRd5bABj8AzanX8Ffvqx77gtVTQa)

File: infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java (hunk 1 above)
Comment (medhavidhawan) [inline infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java]: ok maybe I did not understand the test case correctly. In my conversation with Anny,
she mentioned that the hash for 2 UUIDs (keys) collided and so the streamId did not matter in the scheme. We should check with her once.
@annym could you please provide input.

**Pull request merged**